### PR TITLE
Find slices of root element on slice children

### PR DESCRIPTION
**Pull request by ngfreiter**
This addresses https://standardhealthrecord.atlassian.net/browse/CIMPL-305.

This fixes the issue described in this comment: https://github.com/FHIR/sushi/pull/296#pullrequestreview-381986468. The `findElementByPath` function was unable to find an element when that element was a slice of an element that is itself a child of a slice. This is the FSH that I used to test this out:
```
Profile: FunWithSlicing
Parent: resprate
* category.coding ^slicing.discriminator.path = "$this"
* category.coding ^slicing.discriminator.type = #pattern
* category.coding ^slicing.rules = #open 
* category.coding contains SomeSlice 1..5
* category[VSCat].coding[SomeSlice] ^short = "DO I EXIST"
```
Be aware that to truly test this, you will want to wait until the fixes in https://github.com/FHIR/sushi/pull/325 have been merged. Although this PR does not explicitly depend on that one, the bugs in the IG Publisher that #325 fix will make it difficult to test this PR in the IG Publisher.

### CODE REVIEW

File: src/fhirtypes/StructureDefinition.ts [hunk 1]
```diff
@@ -603,6 +603,26 @@ export class StructureDefinition {
     fisher: Fishable
   ): ElementDefinition {
     let matchingSlice = elements.find(e => e.sliceName === pathPart.brackets.join('/'));
+    if (!matchingSlice && pathPart.brackets?.length === 1) {
+      // If the current element is a child of a slice, the match may exist on the original
+      // sliced element, search for that here
+      for (const e of elements) {
+        const connectedSliceElement = e.findConnectedSliceElement();
+        const matchingConnectedSlice = connectedSliceElement
```
Comment (mint-thompson) [inline src/fhirtypes/StructureDefinition.ts:611]: This `if` block has the same condition as the next `if` block, so I think they can be combined.

File: src/fhirtypes/StructureDefinition.ts [hunk 2]
```diff
@@ -603,6 +603,26 @@ export class StructureDefinition {
...
+          ?.getSlices()
+          .find(e => e.sliceName === pathPart.brackets.join('/'));
+
+        if (matchingConnectedSlice) {
+          const newSlice = matchingConnectedSlice.clone(false);
+          newSlice.id = `${e.id}:${matchingConnectedSlice.sliceName}`;
+          newSlice.structDef = this;
+          e.slicing = connectedSliceElement.slicing;
```
Comment (mint-thompson) [inline src/fhirtypes/StructureDefinition.ts:619]: Is it possible for `e` to already have a defined slicing at this point?

**Comment by ngfreiter** [comment]
Ok @mint-thompson , both your comments should be fixed in the latest commit.

**Comment by mint-thompson** [comment]
I'm still sort of wondering what kind of situation would lead to a case where the existing slicing exists, and is different from `connectedSliceElement.slicing`, but I think it's fine for now.

**Pull request merged**